### ISSUE TITLE
16 bit float use never emits SPV_AMD_gpu_shader_half_float if target-env is vulkan1.1

### ISSUE THREAD

**Issue by krOoze** [KhronosGroup/glslang#1376]
Test shader:
```
#version 450
#extension GL_AMD_gpu_shader_half_float : require

layout (location = 0) in float16_t interpolant;

void main(){
 	// forces SPV_AMD_gpu_shader_half_float to be necessary
	float16_t ac = interpolateAtCentroid(interpolant);
}
```

Output:
```
>glslangValidator.exe -V --target-env vulkan1.1 -H float16.frag
float16.frag
// Module Version 10300
// Generated by (magic number): 80006
// Id's are bound by 12

                              Capability Shader
                              Capability Float16
                              Capability InterpolationFunction
                              Capability StorageInputOutput16
               1:             ExtInstImport  "GLSL.std.450"
                              MemoryModel Logical GLSL450
                              EntryPoint Fragment 4  "main" 10
                              ExecutionMode 4 OriginUpperLeft
                              Source GLSL 450
                              SourceExtension  "GL_AMD_gpu_shader_half_float"
                              Name 4  "main"
                              Name 8  "ac"
                              Name 10  "interpolant"
                              Decorate 10(interpolant) Location 0
               2:             TypeVoid
               3:             TypeFunction 2
               6:             TypeFloat 16
               7:             TypePointer Function 6(float)
               9:             TypePointer Input 6(float)
 10(interpolant):      9(ptr) Variable Input
         4(main):           2 Function None 3
               5:             Label
           8(ac):      7(ptr) Variable Function
              11:    6(float) ExtInst 1(GLSL.std.450) 76(InterpolateAtCentroid) 10(interpolant)
                              Store 8(ac) 11
                              Return
                              FunctionEnd
```

It does not generate `Extension  "SPV_AMD_gpu_shader_half_float"`, which is necessary for `InterpolateAtCentroid` to accept 16-bit floats.

**Comment by johnkslang** [comment]
@amdrexu it seems like the implementation here needs to be finer grained, intercepting operators, not just based on types.

**Comment by krOoze** [comment]
@johnkslang That too (or perhaps the unextended `GLSL.std.450` spec updated -- for some reason it accepts 16-bit for some ops but not others; which I find weird).

In this case though there seem to be outright `if(Vulkan 1.1) do not include SPV_AMD_gpu_shader_half_float` : https://github.com/KhronosGroup/glslang/blob/1831087e4887189f38679adb996437f67f21ebb5/SPIRV/GlslangToSpv.cpp#L2593

**Comment by johnkslang** [comment]
Because Vulkan 1.1 and SPIR-V 1.3 have same native support for some 16 bit. There are multiple ways to get to these now, needing finer-grained care in the logic.

**Comment by amdrexu** [comment]
Yes, I think the logic needs some refining. I will fix it in these two days.

**Comment by amdrexu** [comment]
Addressed by https://github.com/KhronosGroup/glslang/pull/1383

**Comment by arcady-lunarg** [comment]
This issue was fixed by #1383 and should have been closed at that time.
